code/modules/mob/living/basic/space_fauna/cat_surgeon.dm

```diff
@@ -0,0 +1,80 @@
+/**
+ * ## Cat Surgeon
+ *
+ * A mean motherfucker who wants to steal them cat tails... be warned should you see it, as it doesn't really speak much either.
+ */
+
+/mob/living/basic/cat_butcherer
+	name = "Cat Surgeon"
+	desc = "A man with the quest of chasing endless feline tail."
+	icon = 'icons/mob/simple/simple_human.dmi'
+	icon_state = "cat_butcher"
+	icon_living = "cat_butcher"
+	icon_dead = "syndicate_dead"
+	icon_gib = "syndicate_gib"
+	speed = 0.8
+	maxHealth = 100
+	health = 100
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	attack_verb_continuous = "slashes at"
+	attack_verb_simple = "slash at"
+	attack_sound = 'sound/weapons/circsawhit.ogg'
+	combat_mode = TRUE
+	mob_biotypes = MOB_ORGANIC|MOB_HUMANOID
+	sentience_type = SENTIENCE_HUMANOID
+	habitable_atmos = list("min_oxy" = 5, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 1, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_atmos_damage = 7.5
+	faction = list(FACTION_HOSTILE)
+	status_flags = CANPUSH
+	basic_mob_flags = DEL_ON_DEATH
+
+	ai_controller = /datum/ai_controller/basic_controller/cat_butcherer
+
+	/// The stuff we drop on death.
+	var/static/list/drop_on_death = list(
+		/obj/effect/mob_spawn/corpse/human/cat_butcher,
+		/obj/item/circular_saw,
+	)
+
+/mob/living/basic/cat_butcherer/Initialize(mapload)
+	. = ..()
+	apply_dynamic_human_appearance(src, mob_spawn_path = /obj/effect/mob_spawn/corpse/human/cat_butcher, l_hand = /obj/item/circular_saw, bloody_slots = ITEM_SLOT_GLOVES|ITEM_SLOT_OCLOTHING)
+	AddElement(/datum/element/death_drops, drop_on_death)
+	RegisterSignal(src, COMSIG_HOSTILE_POST_ATTACKINGTARGET, PROC_REF(after_attack))
+
+/mob/living/basic/cat_butcherer/Destroy()
+	UnregisterSignal(src, COMSIG_HOSTILE_POST_ATTACKINGTARGET)
+	return ..()
+
+/mob/living/basic/cat_butcherer/proc/after_attack(mob/living/basic/attacker, atom/target)
+	SIGNAL_HANDLER
+
+	if(!iscarbon(target) && prob(65))
+		return // only a 35% chance of dismemberment
```

Comment:
I think your bool logic is messed up. this check passes if the target is a carbon, OR (the target is non carbon AND we pass a 35% chance)
```suggestion
	if(!iscarbon(target) || !prob(35))
		return
```

Reply:
this is another one of those things from the simple animal implementation but i figure a short-circuit will be faster anyhow

Reply:
you short circuited wrong 🗞️ 

x && y && z => !x || !y || !z

Reply:
explained via discord